### PR TITLE
Fix `etcd.status.clusterSize` update

### LINKED ISSUE #255: [BUG] Cluster size calculation in bootstrap case is buggy
**Describe the bug:**
The `etcd.status.cluster.clusterSize` remains empty if the `etcd` creation failed or is interrupted after the `StatefulSet` was created. As a result, the `Ready` condition cannot be calculated:

```
    - lastTransitionTime: "2021-11-05T07:39:42Z"
      lastUpdateTime: "2021-11-05T11:14:17Z"
      message: Cannot determine readiness of cluster since no cluster size has been
        calculated
      reason: ClusterSizeUnknown
      status: Unknown
      type: Ready
```

**Expected behavior:**
`etcd.status.cluster.clusterSize` should be set.

**Environment (please complete the following information):**
 - Etcd-druid version/commit ID :  222a1cc0030846cc137c2aabfd4250cc5172f4ad

### PR DESCRIPTION
**Pull request by timuthy**
**How to categorize this PR?**

/area robustness
/kind bug

**What this PR does / why we need it**:
This PR fixes a referenced bug by making the `etcd.status.clusterSize` update idempotent.

**Which issue(s) this PR fixes**:
Fixes #255

**Release note**:

```bugfix operator
A bug has been fixed which caused the `etcd.status.clusterSize` only being set for new `etcd` resources
```

### CODE REVIEW

**Review by abdasgupta — APPROVED** [review]
Looks good to me.

File: controllers/etcd_controller.go [hunk 1]
```diff
@@ -1519,40 +1510,47 @@ func canDeleteStatefulset(sts *appsv1.StatefulSet, etcd *druidv1alpha1.Etcd) boo
...
 func bootstrapReset(etcd *druidv1alpha1.Etcd) {
 	etcd.Status.Members = nil
-	etcd.Status.ClusterSize = pointer.Int32Ptr(int32(etcd.Spec.Replicas))
+	etcd.Status.ClusterSize = pointer.Int32Ptr(etcd.Spec.Replicas)
 }
 
-func (r *EtcdReconciler) updateEtcdErrorStatus(ctx context.Context, op operationResult, etcd *druidv1alpha1.Etcd, sts *appsv1.StatefulSet, lastError error) error {
+func clusterInBootstrap(etcd *druidv1alpha1.Etcd) bool {
+	return etcd.Status.Replicas == 0 ||
+		(etcd.Spec.Replicas > 1 && etcd.Status.Replicas == 1)
```
Comment (aaronfern) [inline controllers/etcd_controller.go:1518]: Why do we need this check? is it for an upgrade case from single node to multi node?

**Review by aaronfern — COMMENTED** [review]
Looks really good!
Just one query from me, thanks

File: controllers/etcd_controller.go (hunk 1 above)
Comment (timuthy) [reply controllers/etcd_controller.go:1518]: Yes, exactly. According to [this comment](https://github.com/gardener/etcd-druid/pull/188#discussion_r644553365) scaling from 1 -> 3 (or similar) is the same as bootstrapping, i.e. we don't plan to add members via the [members API](https://etcd.io/docs/v3.3/op-guide/runtime-configuration/#add-a-new-member).

Comment (aaronfern) [reply controllers/etcd_controller.go:1518]: Thanks for clarifying

**Review by aaronfern — APPROVED** [review]
/lgtm

**Pull request merged**